Rewrite selection details

MohamedElmdary opened the pull request:
### Description
Rewrite the entire selection details ( node selector [auto/manual], farm selection, location selector, domainName [aka. gateway])
### Changes
- Add `useAsync` hook which allows to handle async calls in an easy way also allow to cancel (ignore result) of async call
- Add `useDeepWatch` fires only if internal keys/values of nested object/array updates (minimize async calls)
- Add `TfSelectionDetails` which group all data from it's internal components
  - `TfAutoSelectNode` used when user select node in *auto* mode and returns `NodeInfo | undefined`
  - `TfManualSelectNode` used when user select node in *manual* mode and returns `NodeInfo | undefined`
  - `TfSelectLocation` used to select location (aka. region and country) returns `{ country?: string, region?: string  } | undefined`
  - `TfSelectFarm` used to select farm returns `FarmInfo | undefined`
  - `TfSelectGpu` used to select gpu cards of a node incase user need *gpu* returns `GPUCardInfo[]`
  - `TfDomainName` used to configure domain if user passed *require-domain* prop returns `DomainInfo | undefined`
- Add some more hints to improve ux
- Integrate SelectionDetails in __full_vm__ solution for testing!
- Integrate SelectionDetails in __ownclound__ solution for testing!

### Related Issues
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1632
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1621
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1635
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1605
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1630
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1312

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

mohamedamer453 commented on packages/playground/src/utils/nodeSelector.ts:
```diff
@@ -0,0 +1,193 @@
+  return {
+    size: options.size,
+    page: Math.max(1, options.page),
+    // availableFor: options.twinId,
```
remove commented line.

MohamedElmdary replied:
Commented for testing because it's very slow in mainnet

0oM4R commented:
i tried to rent a node <141> but it not appears on automatic selection and on manual it gives me that the node dedicated by someone else 
![Screenshot from 2023-12-13 14-25-30](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/d4e8a246-8cae-4205-a3fd-cbe5ca212137)
![Screenshot from 2023-12-13 14-25-15](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/3b244e6e-9358-4173-8c07-8a17ece63f6f)
![Screenshot from 2023-12-13 14-24-45](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/7f317346-f4e8-40c6-b264-1769960b8f11)

mohamedamer453 reviewed (commented):
- Dedicated nodes are listed while they're rented by someone else. (i believe it might be related to the commented `availableFor` lines)
   
   ![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/74624777-b8c3-48b3-8b06-e450494f44e6)

   ![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/69288804-2a63-4e01-ab96-46876b173118)

- The number of loaded nodes is not consistent, sometimes it loads 5 nodes other times it loaded 2 and other times like the cases below it only loaded 1 node, even though there are other nodes to display.

   ![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/60741d39-2544-4ec8-b384-031fd146c862)

- The ids of the loaded nodes are not sorted consistently, sometimes the loaded nodes are sorted in ascending order and sometimes they are not sorted like the case below. 

   ![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/e73cb783-70eb-4176-a212-22575729ba75)
 
- CPU validation isn't working correctly.

   [Screencast from 13-12-23 14:18:12.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/8f37a749-704c-4397-9d93-9e538d70723a)

- SRU validation is broken in the manual selection.

   [Screencast from 13-12-23 14:27:08.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/72406ef3-4d5e-40df-95e6-092eae765959)

- In full VM nodes are loaded partially, 5 nodes and then there's load more nodes button, in other solutions like peertube it's loaded all at once.

   ![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/961de226-97ca-4660-825e-42d5977c3f77)

mohamedamer453 reviewed (commented):
- On mainnet, before selecting or configuring any setting, nothing is shown in the farms dropdown or the nodes one.

   [Screencast from 13-12-23 15:07:49.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/8c5f13e5-92aa-444d-b38a-949ee90a9ecf)

- The automated filter on mainnet doesn't show certified nodes.

   ![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/adf8d8a5-19c0-414d-a93a-ce7868f60ac6)

0oM4R commented:
![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/b9f3e1c5-0d0e-4f03-85fd-424838258492)

can't create an fqdn record 

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/dea8581a-65a3-4b3a-b9de-3b723ca4f688)
![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/315a2b39-c7bd-4eef-86f1-730856600853)
![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/d23c3f4e-1c9e-4009-940c-d357b9c0ce24)

MohamedElmdary commented on packages/playground/src/utils/nodeSelector.ts:
```diff
@@ -0,0 +1,193 @@
+    hasGPU: filters.hasGPU || undefined,
+    rentedBy: filters.dedicated ? options.twinId : undefined,
+    certified: filters.certified || undefined,
+    // availableFor: options.twinId,
```
Commented for testing because it's very slow in mainnet

MohamedElmdary commented:
> * On mainnet, before selecting or configuring any setting, nothing is shown in the farms dropdown or the nodes one.
>   
>     
>       
>       
>   
>       Screencast.from.13-12-23.15.07.49.webm
> * The automated filter on mainnet doesn't show certified nodes.
>   ![image](https://private-user-images.githubusercontent.com/101194226/290199264-adf8d8a5-19c0-414d-a93a-ce7868f60ac6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI0Nzg0MDcsIm5iZiI6MTcwMjQ3ODEwNywicGF0aCI6Ii8xMDExOTQyMjYvMjkwMTk5MjY0LWFkZjhkOGE1LTE5YzAtNDE0ZC1hOTNhLWNlNzg2OGY2MGFjNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIxM1QxNDM1MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMWU4YmYwY2Q0ODFmMmZkZjBmNzM1OGVjODcxNGIxNTEzM2EzMDg5NzQ0ODg2MTkyYzI1NDQ5NDFiZDk3NmFkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.DIcDm5TK4PoK-c2dHCLK4IYn5R8f-gbH9SMhG82jWnw)

Please recheck this issue  and let me know if it still there @mohamedamer453

MohamedElmdary commented:
> * The ids of the loaded nodes are not sorted consistently, sometimes the loaded nodes are sorted in ascending order and sometimes they are not sorted like the case below.
>
> ..

fixed!

MohamedElmdary commented:
> * Dedicated nodes are listed while they're rented by someone else. (i believe it might be related to the commented `availableFor` lines)
>
>
> ...

@mohamedamer453 Please recheck issue should be fixed

mohamedamer453 reviewed (commented):
- If you enter invalid data while using the manual selection and trigger the `Please provide valid data` message and then enter any data in the field the resource validation doesn't work while the deploy buttons stays disabled. and sometimes the deploy button doesn't get disabled.

   [Screencast from 14-12-23 10:53:55.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/9f82952b-0f71-411f-a687-ceef6f100a65)

   ![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/96e6e088-5163-42f7-8a48-ea8619c08682)

   ![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/f91f4b78-ccbe-4eba-aa80-24740075327c)


- Public ip check is broken in the manual node selection, it mentions that there're no available ips while there are in the automated selection.

   [Screencast from 14-12-23 11:09:20.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/756d8cb4-83cf-4731-8265-87dd91f4945e)

mohamedamer453 reviewed (commented):
- Performance on mainnet is still not the best, out of 5 times loading the region/farms/nodes failed 4 times.

   ![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/caf8591d-662d-4728-b5af-ecaac96c4f0e)

- to ensure that the issue isn't only related to mainnet's performance i tried other solutions that didn't have the new selection like micro vm, in micro vm loading the nodes/farms wasn't the fastest but after a while the farms/nodes were loaded every time.

0oM4R commented:
on main net i can't find any certified nodes but using gridproxy there are many of them 
https://gridproxy.grid.tf/nodes?Certification_type=certified


![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/519c75ff-4bd3-4054-8dff-32f2fdd9844f)

commit 796559d4 [development_rewrite_node_selector]

0oM4R commented:
![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/608e52f7-d544-4c35-bb4e-ed9e851b7607)

Should we disable `load nodes` while preparing load farms ?

0oM4R commented:
selected a node and changed the specs, loaded nodes and found the same node matches the new specs, but I have to open the drop-down to re-validate the node.
i suggest to add the validate storage pool to this scenario 
[Screencast from 14 ديس, 2023 EET 01:13:13 م.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/78225c09-5856-4c77-ba7c-0159e1a95d50)

MohamedElmdary commented:
> * If you enter invalid data while using the manual selection and trigger the `Please provide valid data` message and then enter any data in the field the resource validation doesn't work while the deploy buttons stays disabled. and sometimes the deploy button doesn't get disabled.
>   
>     
> ... 

@mohamedamer453 should be fixed

MohamedElmdary commented:
> * Performance on mainnet is still not the best, out of 5 times loading the region/farms/nodes failed 4 times.
>  
>
> ...

Will add retries like old one I think it's time to implement it

MohamedElmdary commented:
> > * Performance on mainnet is still not the best, out of 5 times loading the region/farms/nodes failed 4 times.
> > 
> > ...
> 
> Will add retries like old one I think it's time to implement it

Added tries options and defaults it to 3

MohamedElmdary commented:
> selected a node and changed the specs, loaded nodes and found the same node matches the new specs, but I have to open the drop-down to re-validate the node.
>
>
> ...

fixed

MohamedElmdary commented:
> ![image](https://private-user-images.githubusercontent.com/62248851/290222315-b9f3e1c5-0d0e-4f03-85fd-424838258492.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI1NjkyMjcsIm5iZiI6MTcwMjU2ODkyNywicGF0aCI6Ii82MjI0ODg1MS8yOTAyMjIzMTUtYjlmM2UxYzUtMGQwZS00ZjAzLTg1ZmQtNDI0ODM4MjU4NDkyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE0VDE1NDg0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY3MDM4OTg5YzE1YzkzYzVmYThiYzVjM2JmNzg3NWQxZDhhZGY2MmE0ODFhYTZkMzY1MzI2ODZmZjA2ZDNhNDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6uzyCz429Jh9PpAaYL_K6rtE6EnrI7DOBAc7iLYyk9A)
> 
> can't create an fqdn record
> 
> ![image](https://private-user-images.githubusercontent.com/62248851/290222442-dea8581a-65a3-4b3a-b9de-3b723ca4f688.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI1NjkyMjcsIm5iZiI6MTcwMjU2ODkyNywicGF0aCI6Ii82MjI0ODg1MS8yOTAyMjI0NDItZGVhODU4MWEtNjVhMy00YjNhLWI5ZGUtM2I3MjNjYTRmNjg4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE0VDE1NDg0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThkYTc0ZGQzNDdmYmExMTQzYTAyY2NhMTY0M2E4ZTk2YTgzMmM4MWI5NzdiMGQ2ZmE2OTczNTliOGVmMGZiZmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fLJ1_apXTw54y1WXK1c-NC812JPM4nVE2wbLuNcjrOo) ![image](https://private-user-images.githubusercontent.com/62248851/290222664-315a2b39-c7bd-4eef-86f1-730856600853.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI1NjkyMjcsIm5iZiI6MTcwMjU2ODkyNywicGF0aCI6Ii82MjI0ODg1MS8yOTAyMjI2NjQtMzE1YTJiMzktYzdiZC00ZWVmLTg2ZjEtNzMwODU2NjAwODUzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE0VDE1NDg0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU4OTE5MTExZGZlNWEwNTAzMjlmOWU4Yzc1MDI5ZDY2YjFmYjA3YTM4NjJmNmZiODdkYTY2M2Q1OTUwYTlkNDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.BSk5Jd9OtLIm_x2r9PbbVCt32nNNvvfaR6l0iD1hmpU) ![image](https://private-user-images.githubusercontent.com/62248851/290222735-d23c3f4e-1c9e-4009-940c-d357b9c0ce24.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI1NjkyMjcsIm5iZiI6MTcwMjU2ODkyNywicGF0aCI6Ii82MjI0ODg1MS8yOTAyMjI3MzUtZDIzYzNmNGUtMWM5ZS00MDA5LTk0MGMtZDM1N2I5YzBjZTI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE0VDE1NDg0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTExMmM3MmY5NGVmMGYxNWRlZmU3Y2I4NThhMWJiYjAwMzdmYzA3NGFmMjAyN2VhYjA5Y2Y1ZTJiYTY0YThmY2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.f2o_eGWKkWZOqilQ9W1VXg5XdNZQBBQW4ZCN2pN0K2M)

fixed

0oM4R commented:
> > ![image](https://private-user-images.githubusercontent.com/62248851/290222315-b9f3e1c5-0d0e-4f03-85fd-424838258492.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI1NjkyMjcsIm5iZiI6MTcwMjU2ODkyNywicGF0aCI6Ii82MjI0ODg1MS8yOTAyMjIzMTUtYjlmM2UxYzUtMGQwZS00ZjAzLTg1ZmQtNDI0ODM4MjU4NDkyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE0VDE1NDg0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY3MDM4OTg5YzE1YzkzYzVmYThiYzVjM2JmNzg3NWQxZDhhZGY2MmE0ODFhYTZkMzY1MzI2ODZmZjA2ZDNhNDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6uzyCz429Jh9PpAaYL_K6rtE6EnrI7DOBAc7iLYyk9A)
> > can't create an fqdn record
> > ![image](https://private-user-images.githubusercontent.com/62248851/290222442-dea8581a-65a3-4b3a-b9de-3b723ca4f688.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI1NjkyMjcsIm5iZiI6MTcwMjU2ODkyNywicGF0aCI6Ii82MjI0ODg1MS8yOTAyMjI0NDItZGVhODU4MWEtNjVhMy00YjNhLWI5ZGUtM2I3MjNjYTRmNjg4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE0VDE1NDg0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThkYTc0ZGQzNDdmYmExMTQzYTAyY2NhMTY0M2E4ZTk2YTgzMmM4MWI5NzdiMGQ2ZmE2OTczNTliOGVmMGZiZmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fLJ1_apXTw54y1WXK1c-NC812JPM4nVE2wbLuNcjrOo) ![image](https://private-user-images.githubusercontent.com/62248851/290222664-315a2b39-c7bd-4eef-86f1-730856600853.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI1NjkyMjcsIm5iZiI6MTcwMjU2ODkyNywicGF0aCI6Ii82MjI0ODg1MS8yOTAyMjI2NjQtMzE1YTJiMzktYzdiZC00ZWVmLTg2ZjEtNzMwODU2NjAwODUzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE0VDE1NDg0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU4OTE5MTExZGZlNWEwNTAzMjlmOWU4Yzc1MDI5ZDY2YjFmYjA3YTM4NjJmNmZiODdkYTY2M2Q1OTUwYTlkNDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.BSk5Jd9OtLIm_x2r9PbbVCt32nNNvvfaR6l0iD1hmpU) ![image](https://private-user-images.githubusercontent.com/62248851/290222735-d23c3f4e-1c9e-4009-940c-d357b9c0ce24.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI1NjkyMjcsIm5iZiI6MTcwMjU2ODkyNywicGF0aCI6Ii82MjI0ODg1MS8yOTAyMjI3MzUtZDIzYzNmNGUtMWM5ZS00MDA5LTk0MGMtZDM1N2I5YzBjZTI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE0VDE1NDg0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTExMmM3MmY5NGVmMGYxNWRlZmU3Y2I4NThhMWJiYjAwMzdmYzA3NGFmMjAyN2VhYjA5Y2Y1ZTJiYTY0YThmY2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.f2o_eGWKkWZOqilQ9W1VXg5XdNZQBBQW4ZCN2pN0K2M)
> 
> fixed

i tried to deploy two times but got an error 
![Uploading Screenshot from 2023-12-17 10-43-18.png…]()

AhmedHanafy725 commented on packages/playground/src/stores/grid.ts:
```diff
@@ -7,16 +7,19 @@ import type { Profile } from "./profile_manager";
-      this.grid = profile ? ((await getGrid(profile)) as any) : null;
+      const gridClient = profile ? ((await getGrid(profile)) as any) : null;
+      this.grid = gridClient;
+      this.client = gridClient;
```
why adding this?

MohamedElmdary replied:
name makes more sense
`gridStore.client` vs `gridStor.grid` 
also it doesn't impact any part of project or affect performance

maayarosama commented:
In qa(Twin Id 23):

On choosing node 4 in manual selection, the node is considered valid even though it's rented by twin Id 32.
![Screenshot from 2023-12-17 12-31-07](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/66883096/e813335e-4d85-469e-a289-d01235674c57)

![2023-12-17_12-34](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/66883096/c2997c7a-aae7-4230-9428-ae6390f80265)

AhmedHanafy725 commented:
on devnet selecting any node is showing this error instantly

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/41957921/c5963fb7-4326-4104-ba35-56d1a4a36006)

AhmedHanafy725 commented on packages/grid_client/src/primitives/nodes.ts:
```diff
@@ -366,6 +376,7 @@ class Nodes {
 
   getNodeUrlQuery(options: FilterOptions = {}) {
     const params = {
+      total_cru: options.cru || "",
```
don't forget to add the total cru on the farm query as well

MohamedElmdary commented on packages/grid_client/src/primitives/nodes.ts:
```diff
@@ -340,13 +340,23 @@ class Nodes {
    * @throws Error if there is an issue with the HTTP request or response.
    */
   async getFarmsCount(options: FilterOptions = {}, url = ""): Promise<number> {
+    const _options = { ...options };
```
Because updating object byRef causing rerender in vue

MohamedElmdary commented on packages/playground/src/components/node_selector/TfManualNodeSelector.vue:
```diff
@@ -0,0 +1,181 @@
+          case props.filters.dedicated && node.rentedByTwinId !== gridStore.client.twinId:
+            throw `Node ${nodeId} is Dedicated, but rented by someone else`;
+
+          case props.filters.ipv4 && !node.publicConfig.ipv4:
```
Fix: now loading farm and checking it's public Ips

MohamedElmdary commented on packages/playground/public/config.js:
```diff
@@ -11,4 +11,5 @@ window.env = {
   TFT_ASSET_ISSUER: "GA47YZA3PKFUZMPLQ3B5F2E3CJIB57TGGU7SPCQT2WAEYKN766PWIMB3",
   MINTING_URL: "",
   TIMEOUT: +"10000",
+  PAGE_SIZE: +"5",
```
Updated however it will be reset to 20 while building

0oM4R commented:
> > > ![image](https://private-user-images.githubusercontent.com/62248851/290222315-b9f3e1c5-0d0e-4f03-85fd-424838258492.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI1NjkyMjcsIm5iZiI6MTcwMjU2ODkyNywicGF0aCI6Ii82MjI0ODg1MS8yOTAyMjIzMTUtYjlmM2UxYzUtMGQwZS00ZjAzLTg1ZmQtNDI0ODM4MjU4NDkyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE0VDE1NDg0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY3MDM4OTg5YzE1YzkzYzVmYThiYzVjM2JmNzg3NWQxZDhhZGY2MmE0ODFhYTZkMzY1MzI2ODZmZjA2ZDNhNDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6uzyCz429Jh9PpAaYL_K6rtE6EnrI7DOBAc7iLYyk9A)
> > > can't create an fqdn record
> > > ![image](https://private-user-images.githubusercontent.com/62248851/290222442-dea8581a-65a3-4b3a-b9de-3b723ca4f688.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI1NjkyMjcsIm5iZiI6MTcwMjU2ODkyNywicGF0aCI6Ii82MjI0ODg1MS8yOTAyMjI0NDItZGVhODU4MWEtNjVhMy00YjNhLWI5ZGUtM2I3MjNjYTRmNjg4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE0VDE1NDg0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThkYTc0ZGQzNDdmYmExMTQzYTAyY2NhMTY0M2E4ZTk2YTgzMmM4MWI5NzdiMGQ2ZmE2OTczNTliOGVmMGZiZmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fLJ1_apXTw54y1WXK1c-NC812JPM4nVE2wbLuNcjrOo) ![image](https://private-user-images.githubusercontent.com/62248851/290222664-315a2b39-c7bd-4eef-86f1-730856600853.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI1NjkyMjcsIm5iZiI6MTcwMjU2ODkyNywicGF0aCI6Ii82MjI0ODg1MS8yOTAyMjI2NjQtMzE1YTJiMzktYzdiZC00ZWVmLTg2ZjEtNzMwODU2NjAwODUzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE0VDE1NDg0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU4OTE5MTExZGZlNWEwNTAzMjlmOWU4Yzc1MDI5ZDY2YjFmYjA3YTM4NjJmNmZiODdkYTY2M2Q1OTUwYTlkNDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.BSk5Jd9OtLIm_x2r9PbbVCt32nNNvvfaR6l0iD1hmpU) ![image](https://private-user-images.githubusercontent.com/62248851/290222735-d23c3f4e-1c9e-4009-940c-d357b9c0ce24.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI1NjkyMjcsIm5iZiI6MTcwMjU2ODkyNywicGF0aCI6Ii82MjI0ODg1MS8yOTAyMjI3MzUtZDIzYzNmNGUtMWM5ZS00MDA5LTk0MGMtZDM1N2I5YzBjZTI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE0VDE1NDg0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTExMmM3MmY5NGVmMGYxNWRlZmU3Y2I4NThhMWJiYjAwMzdmYzA3NGFmMjAyN2VhYjA5Y2Y1ZTJiYTY0YThmY2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.f2o_eGWKkWZOqilQ9W1VXg5XdNZQBBQW4ZCN2pN0K2M)
> > 
> > 
> > fixed
> 
> i tried to deploy two times but got an error `failed to setup fqdn proxy: domain already registered.` ![Screenshot from 2023-12-17 10-41-35](https://private-user-images.githubusercontent.com/62248851/291051207-7c941f77-cee3-4803-917b-678dd281144d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI4MTQzOTQsIm5iZiI6MTcwMjgxNDA5NCwicGF0aCI6Ii82MjI0ODg1MS8yOTEwNTEyMDctN2M5NDFmNzctY2VlMy00ODAzLTkxN2ItNjc4ZGQyODExNDRkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE3VDExNTQ1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThjNGIyNjM3YjllZWZjMjc5N2E1Y2JjNDhhYjYyN2Y0MDYyYjk4NzRhYzM0NzNhZGIzZWFjYjg2NDJlODgzNGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.icxVNM6lGWlefy9zYlwshbNAdKe-m_Ma_NT2npvk4ZI) ![Screenshot from 2023-12-17 10-43-18](https://private-user-images.githubusercontent.com/62248851/291051208-62c56f4a-db35-43d2-93cb-815d77cbbcac.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI4MTQzOTQsIm5iZiI6MTcwMjgxNDA5NCwicGF0aCI6Ii82MjI0ODg1MS8yOTEwNTEyMDgtNjJjNTZmNGEtZGIzNS00M2QyLTkzY2ItODE1ZDc3Y2JiY2FjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE3VDExNTQ1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE1N2U5NjY4MWMxMGI1YTNiZGI4ZmIwNjVjOWRmZmYzZjY4MzRjMjgzNTQxZTg3ZjUzZDA5OWYxMzU0YWZhOGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.RcjfnvpzmVYTAJrWKUpqlHhS1rjCIvA735hEfC3dKs4)


worked : 
https://7aga.gridtesting.xyz/login

maayarosama commented:
On mainnet:
There's a difference between the calculation while choosing the same certified node when the certified filter is enabled and disabled.
![2023-12-17_14-59](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/66883096/2091159c-7746-4488-bfee-64a874dcdec5)
![2023-12-17_14-59_1](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/66883096/0253b322-9f9f-4ec1-84ec-eba9bf1c49f0)

amiraabouhadid commented:
> On mainnet: There's a difference between the calculation while choosing the same certified node when the certified filter is enabled and disabled. ![2023-12-17_14-59](https://private-user-images.githubusercontent.com/66883096/291063503-2091159c-7746-4488-bfee-64a874dcdec5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI4MjI4ODAsIm5iZiI6MTcwMjgyMjU4MCwicGF0aCI6Ii82Njg4MzA5Ni8yOTEwNjM1MDMtMjA5MTE1OWMtNzc0Ni00NDg4LWJmZWUtNjRhODc0ZGNkZWM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE3VDE0MTYyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNjN2UzNGJkMGE1MjJjNjlkYWYzZTVmMjMzYWE0NjUxNDJhZTVjZjQyNTIyODdjZTZhYmVmMzkxY2M2NzIyODAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.cVrLfGTLBRyGSnpSY-fZu5fkV4ERu4DzX3lsw1EL3hM) ![2023-12-17_14-59_1](https://private-user-images.githubusercontent.com/66883096/291063505-0253b322-9f9f-4ec1-84ec-eba9bf1c49f0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI4MjI4ODAsIm5iZiI6MTcwMjgyMjU4MCwicGF0aCI6Ii82Njg4MzA5Ni8yOTEwNjM1MDUtMDI1M2IzMjItOWY5Zi00ZWMxLTg0ZWMtZWJhOWJmMWM0OWYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE3VDE0MTYyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyMzhiYjlmMjQ0Nzc0M2NiNDE3ODJiYzZkNjFjYTVlYzQyYTM3NzM1Zjk4NzczNTFhZmZjNjI3N2Q5ZDJmZjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-3Ze9r6e6tIqTg6BWlPQ6Q2dT86KR0YVKULuq39aMps)

this is still the case

MohamedElmdary commented:
> > On mainnet: There's a difference between the calculation while choosing the same certified node when the certified filter is enabled and disabled. ![2023-12-17_14-59](https://private-user-images.githubusercontent.com/66883096/291063503-2091159c-7746-4488-bfee-64a874dcdec5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI4MjI4ODAsIm5iZiI6MTcwMjgyMjU4MCwicGF0aCI6Ii82Njg4MzA5Ni8yOTEwNjM1MDMtMjA5MTE1OWMtNzc0Ni00NDg4LWJmZWUtNjRhODc0ZGNkZWM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE3VDE0MTYyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNjN2UzNGJkMGE1MjJjNjlkYWYzZTVmMjMzYWE0NjUxNDJhZTVjZjQyNTIyODdjZTZhYmVmMzkxY2M2NzIyODAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.cVrLfGTLBRyGSnpSY-fZu5fkV4ERu4DzX3lsw1EL3hM) ![2023-12-17_14-59_1](https://private-user-images.githubusercontent.com/66883096/291063505-0253b322-9f9f-4ec1-84ec-eba9bf1c49f0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI4MjI4ODAsIm5iZiI6MTcwMjgyMjU4MCwicGF0aCI6Ii82Njg4MzA5Ni8yOTEwNjM1MDUtMDI1M2IzMjItOWY5Zi00ZWMxLTg0ZWMtZWJhOWJmMWM0OWYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE3VDE0MTYyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyMzhiYjlmMjQ0Nzc0M2NiNDE3ODJiYzZkNjFjYTVlYzQyYTM3NzM1Zjk4NzczNTFhZmZjNjI3N2Q5ZDJmZjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-3Ze9r6e6tIqTg6BWlPQ6Q2dT86KR0YVKULuq39aMps)
> 
> this is still the case

fixed

MohamedElmdary commented:
- [ ] Make validating configurable
Example
full_vm(disksize) min = 15 and max = 10k but
umber(disk) min = 10 and ...

so we need to implement a way to define validation

0oM4R commented:
after deployment is done, click on `Back` button,  the old node is still selected without checking its resources.

[Screencast from 20 ديس, 2023 EET 11:05:58 ص.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/aa316af4-b069-4ba8-a518-25db0902c445)

[Screencast from 20 ديس, 2023 EET 11:10:57 ص.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/13810f70-970f-464a-96c4-b8d3a1f9938b)

zaelgohary commented:
> after deployment is done, click on `Back` button, the old node is still selected without checking its resources.
> 
>  Screencast.from.20.2023.EET.11.05.58.webm 
>  Screencast.from.20.2023.EET.11.10.57.webm

Issue is opened [here](https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1378)

zaelgohary commented:
@MohamedElmdary  I tried to deploy k8s and got the following error:

[deploy_k8s.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/40770501/90e9b4a1-3c59-40f6-a51c-438917df0630)

zaelgohary commented:
> @MohamedElmdary I tried to deploy k8s and got the following error:
> 
>  deploy_k8s.webm

balance is locked.
